tools/istio-docker.mk

```diff
@@ -87,6 +87,7 @@ docker.proxy_init: $(ISTIO_DOCKER)/istio-iptables.sh
 docker.proxy_init: $(ISTIO_DOCKER)/istio-iptables
 	$(DOCKER_RULE)
 
+docker.sidecar_injector: BUILD_PRE=chmod 755 sidecar-injector &&
```

Comment:
what are the permissions of all the files prior to `chmod`?  This pattern would be better implemented in one place - vs scattared around the docker build operation.  If the permissions are +x, but not gw+x `umask` can bee used to set the default mask.  I think reviewers have been making the assumption the permissions lack +x.  Can you expand a bit on why this chmod 755 is needed?

Cheers
-steve

Reply:
I have tested in local, the permission is already 755. but there is similar problem here - https://github.com/istio/istio/pull/13579 after removed the chmod 755, it throws `permission denied`. that is why the envoy and pilot-agent add such chmod line.